### ISSUE TITLE
PHP 7.0, 7.1, 7.2, 7.3 source packages - Patch discrepancy (patch for build timestamp removal)

### ISSUE THREAD

**Issue by nuxwin** [oerdnj/deb.sury.org#976]
Good morning @oerdnj

I think that I did found a big issue with PHP **7.0**, **7.1**, **7.2** and **7.3** source packages.

In PHP **5.6** source package, there is a patch named `0041-Add-patch-to-remove-build-timestamps-from-generated-.patch`. We agree on the fact that the purpose of this patch is to remove the build timestamp from generated files. So, the PHP version output should look as follows:

```
PHP 5.6.38-2+0~20181015120552.6+jessie~1.gbp567807 (cli) 
Copyright (c) 1997-2016 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2016, by Zend Technologies
```

Problem is that with all other PHP versions, the output look as follows (example for PHP 7.2):

```
PHP 7.2.11-2+0~20181015120510.9+jessie~1.gbp8105e0 (cli) (built: Oct 16 2018 03:39:58) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.11-2+0~20181015120510.9+jessie~1.gbp8105e0, Copyright (c) 1999-2018, by Zend Technologies
```
As you can see there, the build timestamp is back... 
In fact, in the PHP 7.0, 7.1, 7.2, and 7.3 source packages, the patch is also present but if you look at it, it doesn't do what it is expected to do. For instance, for PHP 7.2, it look like:

```diff
From: Thijs Kinkhorst <thijs@debian.org>
Date: Wed, 15 Jun 2016 09:18:03 +0200
Subject: Add patch to install php7 module directly to APXS_LIBEXEC

---
 sapi/apache2handler/config.m4 | 15 +++------------
 1 file changed, 3 insertions(+), 12 deletions(-)

diff --git a/sapi/apache2handler/config.m4 b/sapi/apache2handler/config.m4
index f286b1a..fba92b8 100644
--- a/sapi/apache2handler/config.m4
+++ b/sapi/apache2handler/config.m4
@@ -59,18 +59,9 @@ if test "$PHP_APXS2" != "no"; then
   APACHE_CFLAGS="$APACHE_CPPFLAGS -I$APXS_INCLUDEDIR $APR_CFLAGS $APU_CFLAGS -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1"
 
   APXS_LIBEXECDIR='$(INSTALL_ROOT)'`$APXS -q LIBEXECDIR`
-  if test -z `$APXS -q SYSCONFDIR`; then
-    INSTALL_IT="\$(mkinstalldirs) '$APXS_LIBEXECDIR' && \
-                 $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
-                       -i -n php7"
-  else
-    APXS_SYSCONFDIR='$(INSTALL_ROOT)'`$APXS -q SYSCONFDIR`
-    INSTALL_IT="\$(mkinstalldirs) '$APXS_LIBEXECDIR' && \
-                \$(mkinstalldirs) '$APXS_SYSCONFDIR' && \
-                 $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
-                       -S SYSCONFDIR='$APXS_SYSCONFDIR' \
-                       -i -a -n php7"
-  fi
+  INSTALL_IT="\$(mkinstalldirs) '$APXS_LIBEXECDIR' && \
+               $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
+                     -i -n php7"
 
   case $host_alias in
   *aix*)
```
As you can see, the purpose is totallly different... My thinking is that you have done a mistake during a patch rebase or something similar, replacing the patch content with content of another patch (which you wanted add?).

Thank you for your time.

**BTW:** Whats was the intent for build timestamp removal?

**Comment by nuxwin** [comment]
@oerdnj 

I did found similar patch which holds changes for both removal of build timestamp and installation of php7 module directly to APXS_LIBEXEC

Now, why those changes (which target different purpose) are provided in the same patch is a mistery for me.

Thank you for your interest.

**Comment by oerdnj** [comment]
I investigated and it's actually fine on PHP 7.1+, as `__DATE__` and `__TIME__` is handled automatically and `PHP_BUILD_DATE` UTC patch from Chris Lamb was merged upstream.

So, I fixed `PHP_BUILD_DATE` on PHP 7.0 (not strictly needed in the PPA though) and mainly I fixed the wrong patch name.
